### PR TITLE
4 seasons now works

### PR DESCRIPTION
**Pull request by lnzd17**
PR for #570 
* dogs that have a name with a number at the beginning can now be searched

### CODE REVIEW

File: app/models/dog_searcher.rb [hunk 1]
```diff
@@ -71,7 +71,9 @@ def self.search(params: {}, manager: false)
   private
 
   def tracking_id_search?
```
Comment (TheDudeWithTheThing) [inline app/models/dog_searcher.rb]: well what are we really trying to test here?  the whole string is a number?  The problem is that `.to_i` will parse for any number and return true now.

```
[1] pry(main)> "4 seasons".to_i
=> 4
```

maybe the real fix is to make sure the whole string is numbers

```
return @params[:search].to_s.match(/^[0-9]+$/)
```

http://rubular.com/r/Z89CWp7QdZ

Comment (TheDudeWithTheThing) [inline app/models/dog_searcher.rb]: Also I just learned that `===` when the regex is the first argument returns true / false, good stuff!

Comment (lnzd17) [inline app/models/dog_searcher.rb]: 👍 makes more sense to check for 'only numbers?' instead of checking for 'any letters?'. I will revise. Thanks!

Comment (markottaviani) [inline app/models/dog_searcher.rb]: I'll try and do a query of the production database tonight and see what the Microchip IDs actually look like.   I know several brands are in use and I kind of forget what the formats look like.  There is a chance that some are a blend of letters and numbers.

Comment (markottaviani) [inline app/models/dog_searcher.rb]: And of course now I see that people sometimes put spaces in the IDs...

Comment (lnzd17) [inline app/models/dog_searcher.rb]: haha of course. I will hold off with updates for now.

Comment (markottaviani) [inline app/models/dog_searcher.rb]: Okay, looks like letters sometime show up in the microchip values as well.   Maybe something like this for a different approach?  https://stackoverflow.com/questions/9379884/search-multiple-columns-rails

Comment (lnzd17) [inline app/models/dog_searcher.rb]: I added some changes considering the link above from @markottaviani.

**Comment by lnzd17** [comment]
@markottaviani  Just wanted to check in on this PR and see if I should make any other changes. Thanks! :)

**Comment by markottaviani** [comment]
Thanks for the reminder, I should have a chance to review this tonight.  @TheDudeWithTheThing LMK if you have any feedback.

**Comment by TheDudeWithTheThing** [comment]
I'm good  👍

**Pull request merged**